cmd/admin-monitor.go

```diff
@@ -0,0 +1,252 @@
+	for i := range s.CPULoad.Load {
+		msg += fmt.Sprintf("   current    %.2f%%      %.2f%%    %.2f%%\n", s.CPULoad.Load[i].Min, s.CPULoad.Load[i].Avg, s.CPULoad.Load[i].Max)
+		if len(s.CPULoad.HistoricLoad) > i {
+			msg += fmt.Sprintf("   historic   %.2f%%      %.2f%%    %.2f%%\n", s.CPULoad.HistoricLoad[i].Min, s.CPULoad.HistoricLoad[i].Avg, s.CPULoad.HistoricLoad[i].Max)
+		}
+		msg += "\n"
+	}
+
+	// Mem section
+	msg += fmt.Sprintf("%s        usage\n", console.Colorize(monitor, "   MEM"))
+	for i := range s.MemUsage.Usage {
+		msg += fmt.Sprintf("   current    %s\n", humanize.IBytes(s.MemUsage.Usage[i].Mem))
+		if len(s.MemUsage.HistoricUsage) > i {
+			msg += fmt.Sprintf("   historic   %s\n", humanize.IBytes(s.MemUsage.HistoricUsage[i].Mem))
+		}
+		msg += "\n"
+	}
+
+	return msg
+}
+
+func mainAdminMonitor(ctx *cli.Context) error {
+	checkAdminMonitorSyntax(ctx)
+
+	// set the console colors
+	console.SetColor(monitor, color.New(color.FgGreen, color.Bold))
+	console.SetColor(monitorDegraded, color.New(color.FgYellow, color.Bold))
+	console.SetColor(monitorFail, color.New(color.FgRed, color.Bold))
+
+	// Get the alias
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	// Create a new minio admin client
+	client, err := newAdminClient(aliasedURL)
+	fatalIf(err, "Cannot get a configured admin connection.")
+
+	printOfflineErrorMessage := func(err error) {
+		errMsg := ""
+		if err != nil {
+			errMsg = err.Error()
+		}
+		printMsg(serverMonitorMessage{
+			Addr:    aliasedURL,
+			Service: "off",
+			Err:     errMsg,
+		})
+	}
+
+	processErr := func(e error) error {
+		switch e.(type) {
+		case *json.SyntaxError:
+			printOfflineErrorMessage(e)
+			return e
+		case *url.Error:
+			printOfflineErrorMessage(e)
+			return e
+		default:
+			// If the error is not nil and unrecognized, just print it and exit
+			fatalIf(probe.NewError(e), "Cannot get service status.")
+		}
+		return nil
+	}
+	// Fetch info of all CPU loads (all minio server instances)
+	cpuLoads, e := client.ServerCPULoadInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately if error encountered
+		return nil
+	}
+
+	sort.Stable(&sortWrapper{cpuLoads: cpuLoads})
+
+	memUsages, e := client.ServerMemUsageInfo()
+	if err := processErr(e); err != nil {
+		// exit immediately if error encountered
+		return nil
+	}
+	sort.Stable(&sortWrapper{memUsages: memUsages})
+
+	for i, cpuLoad := range cpuLoads {
+		memUsage := memUsages[i]
+		if cpuLoad.Error != "" {
+			printMsg(serverMonitorMessage{
+				Service: "off",
+				Addr:    cpuLoad.Addr,
+				Err:     cpuLoad.Error,
+			})
+			continue
+		}
+		if memUsage.Error != "" {
+			printMsg(serverMonitorMessage{
+				Service: "off",
+				Addr:    memUsage.Addr,
+				Err:     memUsage.Error,
+			})
+			continue
+		}
+
+		printMsg(serverMonitorMessage{
+			Service:  "on",
+			Addr:     cpuLoad.Addr,
+			CPULoad:  &cpuLoad,
+			MemUsage: &memUsage,
+		})
+	}
+	return nil
+}
+
+type sortWrapper struct {
```

Comment:
We should order them by their occurrence in the endpoints list.
 I.e in `minio server http://node1/disk1 http://node2/disk2 ...`, `node1` statistics should come before `node2` and so on, left to right.

Reply:
Why are we sorting the list of CPU/memory usage of minio servers in ascending order of the server address? What is the rationale? (Note: the corresponding code change is no longer part of this PR)